File: Assets/Scripts/Keys.js
#pragma strict
public var BlockLock : GameObject;
var boi = false;

function Start () {
	Time.timeScale = PlayerPrefs.GetInt("paused");
}

function OnCollisionEnter(col : Collision) {
	
	if ((col.collider.name == "Sphere" || col.collider.name == "Sphere #2") && boi == false) {
		BlockLock.SetActive (false);
		boi = true;
	}else if ((col.collider.name == "Sphere" || col.collider.name == "Sphere #2") && boi == true) {
		BlockLock.SetActive (true);
		boi = false;
	}
}

File: Assets/Scripts/Keys.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class Keys : MonoBehaviour
{
    public GameObject BlockLock;
    public bool boi;
    public virtual void Start()
    {
        Time.timeScale = PlayerPrefs.GetInt("paused");
    }

    public virtual void OnCollisionEnter(Collision col)
    {
        if (((col.collider.name == "Sphere") || (col.collider.name == "Sphere #2")) && (this.boi == false))
        {
            this.BlockLock.SetActive(false);
            this.boi = true;
        }
        else
        {
            if (((col.collider.name == "Sphere") || (col.collider.name == "Sphere #2")) && (this.boi == true))
            {
                this.BlockLock.SetActive(true);
                this.boi = false;
            }
        }
    }

}